src/main/java/io/jenkins/plugins/jfrog/JfStep.java

```diff
@@ -207,32 +207,43 @@ public Launcher.ProcStarter setupJFrogEnvironment(Run<?, ?> run, EnvVars env, La
          * does not support stdin input, because it is a custom launcher.
-         *
-         * @param workspace The workspace file path.
-         * @param env       The environment variables.
-         * @param launcher  The command launcher.
+         * @param environmentVariables The environment variables.
          * @return true if stdin-based password handling is supported; false otherwise.
          */
-        public boolean isPasswordStdinSupported(FilePath workspace, EnvVars env, Launcher launcher, String jfrogBinaryPath) throws IOException, InterruptedException {
+        public boolean isPasswordStdinEnabled(FilePath workspace, EnvVars environmentVariables, Launcher launcher, String jfrogBinaryPath) throws IOException, InterruptedException {
             TaskListener listener = getContext().get(TaskListener.class);
             JenkinsBuildInfoLog buildInfoLog = new JenkinsBuildInfoLog(listener);
-
-            boolean isPluginLauncher = launcher.getClass().getName().contains("org.jenkinsci.plugins");
-            if (isPluginLauncher) {
-                buildInfoLog.debug("Password stdin is not supported,Launcher is a plugin launcher.");
-                return false;
-            }
-            Launcher.ProcStarter procStarter = launcher.launch().envs(env).pwd(workspace);
+            String readJFrogCliPwdStdinSupport = environmentVariables.get("JFROG_CLI_PASSWORD_STDIN_SUPPORT", "");
+            Launcher.ProcStarter procStarter = launcher.launch().envs(environmentVariables).pwd(workspace);
             Version currentCliVersion = getJfrogCliVersion(procStarter, jfrogBinaryPath);
-            buildInfoLog.debug("Password stdin is supported");
-            return currentCliVersion.isAtLeast(MIN_CLI_VERSION_PASSWORD_STDIN);
+            boolean isMinimumCLIVersionPasswdSTDIN = currentCliVersion.isAtLeast(MIN_CLI_VERSION_PASSWORD_STDIN);
+            if (StringUtils.isBlank(readJFrogCliPwdStdinSupport)) {
+                boolean isPluginLauncher = launcher.getClass().getName().contains("org.jenkinsci.plugins");
+                if (isPluginLauncher) {
+                    buildInfoLog.debug("Password stdin is not supported,Launcher is a plugin launcher.");
+                    return false;
+                }
+                buildInfoLog.debug("Password stdin is supported");
+                return isMinimumCLIVersionPasswdSTDIN;
+            }
+            boolean isSupported = Boolean.parseBoolean(readJFrogCliPwdStdinSupport);
+            if (!isMinimumCLIVersionPasswdSTDIN && isSupported) {
+                buildInfoLog.error("Password input via stdin is not supported, JFrog CLI version is below the minimum required version.");
+            } else if (isMinimumCLIVersionPasswdSTDIN && isSupported) {
+                return true;
+            }
+            buildInfoLog.debug("Password input via stdin is not supported, Launcher is a plugin launcher.");
+            return false;
```

Comment:
```suggestion
    boolean isSupported = Boolean.parseBoolean(readJFrogCliPwdStdinSupport);
    if (isSupported && !isMinimumCLIVersionPasswdSTDIN) {
        buildInfoLog.error("Password input via stdin is not supported, JFrog CLI version is below the minimum required version.");
    }
    return isSupported && isMinimumCLIVersionPasswdSTDIN;
```

Can be simplified a bit